src/libraries/Microsoft.PowerFx.LanguageServerProtocol/Handlers/Nl2Fx/Nl2FxLanguageServerOperationHandler.cs

```diff
@@ -53,7 +53,8 @@ private void PreHandleNl2Fx(IPowerFxScope scope, LanguageServerOperationContext
             {
                 Sentence = _nl2FxRequestParams.Sentence,
                 SymbolSummary = summary,
-                Engine = check.Engine
+                Engine = check.Engine,
+                ExpressionCultureInfo = check.ParserCultureInfo

```

Comment:
What if we pass `CheckResult` only?

Reply:
Do you mean when ParserCultureInfo is null? then we just have locale null and we handle it in our NL library later.

Reply:
No! we should avoid exposing check results out of this function. This function runs behind a host (document server for example) lock and check result holds references to host's not thread safe data structures. Just get what you need from check result and pass it through this object. don't throw out the whole check result from here

Reply:
The idea was not exposing the whole object. We would pass `CheckResult` and then call something like `public Engine Engine => checkResult.Engine`. Just a suggestion. Whatever works best.

Reply:
I think Akshar suggest to not pass checkresult here so i keep this as it is to safe.

Reply:
yes, least privilege. 
Just culture is good here.